library/std/src/sys/unix/stack_overflow.rs

```diff
@@ -143,14 +143,15 @@ mod imp {
     }
 
     unsafe fn get_stackp() -> *mut libc::c_void {
-        let stackp = mmap(
-            ptr::null_mut(),
-            SIGSTKSZ + page_size(),
-            PROT_READ | PROT_WRITE,
-            MAP_PRIVATE | MAP_ANON,
-            -1,
-            0,
-        );
+        // OpenBSD requires this flag for stack mapping
+        // otherwise the said mapping will fail is a no-op in most systems
+        // and has a different meaning on FreeBSD
+        #[cfg(target_os = "openbsd")]
+        let flags = MAP_PRIVATE | MAP_ANON | libc::MAP_STACK;
+        #[cfg(not(target_os = "openbsd"))]
+        let flags = MAP_PRIVATE | MAP_ANON;
```

Comment:
Linux supports `MAP_STACK` too. It is a no-op on all architectures. FreeBSD supports it too. For example macOS doesn't support it though.

Reply:
yes but not necessary on these oses plus solaris/illumos does not have it and I believe haiku neither.

Reply:
> yes but not necessary

Not **currently** necessary. A future architecture may need it. That is why it exists as no-op at all.

Reply:
ok can add for linux, netbsd but I m less sure about FreeBSD the flag is working but does not seem to work well with null start addresses, seems little usage difference.